### PR TITLE
gyb: 1.80 -> 1.81

### LINKED ISSUE #298925: Build failure: gyb
### Steps To Reproduce

1. build gyb

Seems to fail both on master and on latest unstable.

### Build log

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/j62bpk1j28vpdgw3gh6n6lakyzvvc987-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/windows-gyb.wxs
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
no Makefile or custom buildPhase, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: installPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80
checking for references to /build/ in /nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80...
patching script interpreter paths in /nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80
/nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80/bin/gyb: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/7wz6hm9i8wljz0hgwz1wqmn2zlbgavrq-python3-3.11.8/bin/python3"
stripping (with command strip and flags -S -p) in  /nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80/lib /nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80/bin
Rewriting #!/nix/store/7wz6hm9i8wljz0hgwz1wqmn2zlbgavrq-python3-3.11.8/bin/python3 to #!/nix/store/7wz6hm9i8wljz0hgwz1wqmn2zlbgavrq-python3-3.11.8
wrapping `/nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80/bin/gyb'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Running phase: installCheckPhase
Traceback (most recent call last):
  File "/nix/store/j3axr9ynpbhwgpg2ax0hqvxqvq91c0h7-gyb-1.80/bin/.gyb-wrapped", line 87, in <module>
    import google.oauth2.service_account
ModuleNotFoundError: No module named 'google.oauth2'

```

### Notify maintainers

@austinbutler

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.8.1-cachyos, NixOS, 24.05 (Uakari), 24.05.20240323.44d0940`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/amxd2p02wx78nyaa4bkb0hjvgwhz1dq7-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by austinbutler**
## Description of changes

https://github.com/GAM-team/got-your-back/compare/v1.80...v1.81

Fixes https://github.com/NixOS/nixpkgs/issues/298925.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/backup/gyb/default.nix [hunk 1]
```diff
@@ -28,14 +27,19 @@ python3Packages.buildPythonApplication rec {
...
+    mv gyb.py $out/bin/gyb
+    mv *.py $out/${python3.sitePackages}/
 
     runHook postInstall
   '';
 
   checkPhase = ''
+    runHook preCheck
+
+    unset PYTHONPATH
```
Comment (austinbutler) [inline pkgs/tools/backup/gyb/default.nix:39]: Main fix for the build is this. No idea why this is now necessary when it worked before.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 307668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gyb</li>
  </ul>
</details>

File: pkgs/tools/backup/gyb/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/backup/gyb/default.nix:39]: We should move that into a subshell, to not effect other hooks.

Comment (austinbutler) [reply pkgs/tools/backup/gyb/default.nix]: Makes sense, changed.

File: pkgs/tools/backup/gyb/default.nix [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-, python3Packages
 }:
 
-python3Packages.buildPythonApplication rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "gyb";
-  version = "1.80";
-  format = "other";
+  version = "1.81";
+  pyproject = false;
```
[the remark this answers is not in the document]
Comment (austinbutler) [reply pkgs/tools/backup/gyb/default.nix]: I got the opposite impression from [the docs](https://github.com/NixOS/nixpkgs/blob/03d792fbdfc59fc813c7abb2fbad0dcc3b33a9d4/doc/languages-frameworks/python.section.md) but reverted this change anyway.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 307668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gyb</li>
  </ul>
</details>

**Review by nadir-ishiguro — APPROVED** [review]

**Comment by nadir-ishiguro** [comment]
Result of `nixpkgs-review pr 307668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gyb</li>
  </ul>
</details>

**Pull request merged**